lib/internal/streams/pipeline.js

```diff
@@ -258,6 +258,11 @@ function pipeline(...streams) {
       throw new ERR_INVALID_ARG_TYPE(
         name, ['Stream', 'Function'], stream);
     }
+
+    if (isStream(ret) && ret.destroyed) {
+      finish(new ERR_STREAM_DESTROYED(reading ? 'pipe' : 'write'));
```

Comment:
Why not `ERR_STREAM_PREMATURE_CLOSE` so it would be consistent with what will happen when the stream is destroyed during `pipeline`?